scipy/sparse/base.py

```diff
@@ -249,7 +249,7 @@ def dot(self, other):
         array([ 1, -3, -1], dtype=int64)
 
         """
-        return self * other
+        return self.__mul__(other)
```

Comment:
I think this is incorrect in principle --- `dot` should not return NotImplemented.